vcr/pe/store.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package pe
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+)
+
+// Store is a store for presentation definitions
+// It loads a file with the mapping from oauth scope to presentation definition
+type Store struct {
+	// mapping holds the oauth scope to presentation definition mapping
+	mapping map[string]PresentationDefinition
+}
+
+// LoadFromFile loads the mapping from the given file
+func (s *Store) LoadFromFile(filename string) error {
+	// read the bytes from the file
+	reader, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+	bytes, err := io.ReadAll(reader)
+	if err != nil {
+		return err
+	}
+
+	return s.LoadFromBytes(bytes)
+}
+
+func (s *Store) LoadFromBytes(bytes []byte) error {
```

Comment:
unused outside of test code in this package, so make it non-exported

Reply:
I planned on using it somewhere, but wasn't needed so removed.